### PR TITLE
installation changes

### PR DESCRIPTION
**Pull request by larryk85**
Added an install and uninstall script and a eosio.version.hpp.  In addition to these, this PR also adds some generated cmake files to be used by external projects like eosio.contracts to resolve dependencies more easily.  This also allows for an external project to build against an uninstalled eos.

### CODE REVIEW

**Comment by lazaridiscom** [comment]
@larryk85 , does this fix the issue mention in https://github.com/EOSIO/eos/issues/4883#issuecomment-408205732 by @jgiszczak ?

**Comment by larryk85** [comment]
@lazaridiscom and @jgiszczak yes this does fix that issue.

File: eosio_build.sh [hunk 1]
```diff
@@ -52,7 +52,7 @@
 	DISK_MIN=20
 	DOXYGEN=false
 	ENABLE_COVERAGE_TESTING=false
-	CORE_SYMBOL_NAME="SYS"
+	CORE_SYMBOL_NAME="EOS"
```
Comment (heifner) [inline eosio_build.sh]: Looks like this was changed by mistake.

Comment (larryk85) [reply eosio_build.sh]: Didn't mean to push this here, reverting.

**Pull request merged**